### PR TITLE
Always load local clerk-js when running integration tests

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description
- Add `http-server` as a dep
- Promisify `treekill` to correctly await it
- Serve locally-built clerk-js files using http-server
- Install and serve clerk-js from verdaccio on CICD

## Checklist

- [x] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [x] `build/tooling/chore`

### CODE REVIEW

**Review by dimkl — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -124,8 +124,12 @@ jobs:
         working-directory: ./integration
         run: npm init -y && npm install @clerk/backend
 
+      - name: Install @clerk/clerk-js in os temp
+        working-directory: ${{runner.temp}}
+        run: mkdir clerk-js && cd clerk-js && npm init -y && npm install @clerk/clerk-js
+
       - name: Run Integration Tests
-        run: npm run test:integration:${{ matrix.test-name }}
+        run: E2E_APP_CLERK_JS_DIR=$RUNNER_TEMP npm run test:integration:${{ matrix.test-name }}
```
Comment (LekoArts) [inline .github/workflows/ci.yml:132]: Nit: Define the env var in the `env` section below that script

File: integration/models/application.ts [hunk 2]
```diff
@@ -63,9 +61,10 @@ export const application = (config: ApplicationConfig, appDirPath: string, appDi
         stderr: opts.detached ? fs.openSync(stderrFilePath, 'a') : undefined,
         log: opts.detached ? undefined : log,
       });
+      // const shouldRetry = () => proc.exitCode !== 0 && proc.exitCode !== null;
```
Comment (LekoArts) [inline integration/models/application.ts:64]: Nit: Dead comment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nikosdouvlis) [reply .github/workflows/ci.yml:132]: Opening a second PR, thanks @LekoArts that's cleaner

File: integration/models/application.ts (hunk 2 above)
Comment (nikosdouvlis) [reply integration/models/application.ts:64]: This should've been a todo - changing it in the new PR

**Pull request merged**